src/common/snippets/src/lowered/pass/move_scalar_to_consumer.cpp

```diff
@@ -26,9 +26,17 @@ bool MoveScalarToConsumer::run(LinearIR& linear_ir) {
         const auto expr = expr_it->get();
         if (ov::is_type<op::Scalar>(expr->get_node())) {
             const auto consumers = expr->get_output_port_connector(0)->get_consumers();
-            OPENVINO_ASSERT(consumers.size() == 1, "Scalar expression is expected to have a single consumer");
+            OPENVINO_ASSERT(!consumers.empty(), "Scalar expression should have at least one consumer");
+            const auto loop_ids = consumers.begin()->get_expr()->get_loop_ids();
+            for (const auto& consumer : consumers) {
+                OPENVINO_ASSERT(consumer.get_expr()->get_loop_ids() == loop_ids,
+                                "All consumers of a Scalar expression are expected to have the same loop IDs");
+            }
+            auto consumer_expr = std::min_element(consumers.begin(), consumers.end(),
+                [](const ExpressionPort& lhs, const ExpressionPort& rhs) {
+                    return lhs.get_expr()->get_exec_num() < rhs.get_expr()->get_exec_num();
+                })->get_expr();
```

Comment:
Minor: instead of doubled iterations through `consumers`, we can unite them into one: we will validate `loop_ids` and find the earliest expr at the same time. But I don't insist :)

Reply:
Applied